frontend/__tests__/RequestContext.test.jsx

```
@@ -0,0 +1,236 @@
+
+// Test component to consume the context
+const TestConsumer = () => {
+  const { requestCount, setRequestCount } = useContext(RequestContext);
+  
+  return (
+    <div>
+      <div data-testid="request-count">{requestCount}</div>
+      <button 
+        data-testid="increment-button" 
+        onClick={() => setRequestCount(prev => prev + 1)}
+      >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)